src/valhalla_associate_segments.cc

```diff
@@ -69,58 +71,37 @@ enum class MatchType : uint8_t {
   kShortestPath = 1
 };
 
-struct MatchStats {
-  uint32_t success_count;
-  uint32_t failure_count;
-  uint32_t walk_count;
-  uint32_t path_count;
-
-  MatchStats()
-    : success_count(0),
-      failure_count(0),
-      walk_count(0),
-      path_count(0) {
+// Matched edge
+struct EdgeMatch {
+  GraphId edgeid;
+  uint32_t length;
+  float start_pct;
+  float end_pct;
+
+  EdgeMatch(const GraphId& e, const uint32_t l, const float start, const float end)
+    : edgeid(e),
+      length(l),
+      start_pct(start),
+      end_pct(end) {
   }
 };
 
-vm::PointLL interp(vm::PointLL a, vm::PointLL b, double frac) {
-  return vm::PointLL(a.AffineCombination(1.0 - frac, frac, b));
-}
-
-// chop the first "dist" length off seg, returning it as the result. this will
-// modify seg!
-std::vector<vm::PointLL> chop_subsegment(std::vector<vm::PointLL> &seg, uint32_t dist) {
-  const size_t len = seg.size();
-  assert(len > 1);
-
-  std::vector<vm::PointLL> result;
-  result.push_back(seg[0]);
-  double d = 0.0;
-  size_t i = 1;
-  for (; i < len; ++i) {
-    auto segdist = seg[i-1].Distance(seg[i]);
-    if ((d + segdist) >= dist) {
-      double frac = (dist - d) / segdist;
-      auto midpoint = interp(seg[i-1], seg[i], frac);
-      result.push_back(midpoint);
-      // remove used part of seg.
-      seg.erase(seg.begin(), seg.begin() + (i - 1));
-      seg[0] = midpoint;
-      break;
+// Candidate edges for the start and end. Includes the path edge plus
+// distance from segment start/end
+struct CandidateEdge {
+  vb::PathLocation::PathEdge edge;
+  float distance;
 
-    } else {
-      d += segdist;
-      result.push_back(seg[i]);
-    }
+  CandidateEdge()
```

Comment:
if you remove these constructors this will do what you want just by doing `CandidateEdge{}`